### ISSUE TITLE
[package] Docker does not honor explicit <imageName> setting

### ISSUE THREAD

**Issue by aalmiray** [jreleaser/jreleaser#952]
### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [ ] Example that reproduces the problem (link to git repository is ideal)
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Clone https://gitlab.com/myron0815/jreleaser
2. Make sure to have a Docker deamon running.
3. Build the project with `mvn verify`

### Expected Behaviour

A successful docker container image.

### Actual Behaviour

Failure due to invalid tag.

### Environment Information

- **Operating System**: N/A
- **JReleaser Version:** 1.2.0
- **JDK Version:** N/A

```
[DEBUG]     [docker] docker -l error build --no-cache -q -f /tmp/gitlab/jreleaser/target/jreleaser/package/myapp/docker/Dockerfile -t gitlabRepoOwner/myapp:GIT /tmp/gitlab/jreleaser/target/jreleaser/package/myapp/docker
[INFO]      [docker]  - gitlabRepoOwner/myapp:GIT
[ERROR]     [docker] invalid argument "gitlabRepoOwner/myapp:GIT" for "-t, --tag" flag: invalid reference format: repository name must be lowercase
[ERROR]     [docker] See 'docker build --help'.
[WARN]      [docker] failure: Unexpected error
org.jreleaser.model.packager.spi.PackagerProcessingException: Unexpected error
	at org.jreleaser.packagers.AbstractPackagerProcessor.executeCommand(AbstractPackagerProcessor.java:241)
	at org.jreleaser.packagers.DockerPackagerProcessor.packageDocker(DockerPackagerProcessor.java:201)
	at org.jreleaser.packagers.DockerPackagerProcessor.doPackageDistribution(DockerPackagerProcessor.java:157)
	at org.jreleaser.packagers.AbstractTemplatePackagerProcessor.doPackageDistribution(AbstractTemplatePackagerProcessor.java:138)
	at org.jreleaser.packagers.AbstractPackagerProcessor.packageDistribution(AbstractPackagerProcessor.java:161)
	at org.jreleaser.packagers.DockerPackagerProcessor.packageDistribution(DockerPackagerProcessor.java:62)
	at org.jreleaser.engine.distribution.DistributionProcessor.packageDistribution(DistributionProcessor.java:108)
	at org.jreleaser.engine.distribution.Distributions.processPackager(Distributions.java:126)
	at org.jreleaser.engine.distribution.Distributions.processDistribution(Distributions.java:103)
	at org.jreleaser.engine.distribution.Distributions.process(Distributions.java:89)
	at org.jreleaser.workflow.PackageWorkflowItem.doInvoke(PackageWorkflowItem.java:37)
	at org.jreleaser.workflow.AbstractWorkflowItem.lambda$invoke$0(AbstractWorkflowItem.java:38)
	at org.jreleaser.engine.hooks.CommandHookExecutor.execute(CommandHookExecutor.java:52)
	at org.jreleaser.workflow.AbstractWorkflowItem.invoke(AbstractWorkflowItem.java:38)
	at org.jreleaser.workflow.WorkflowImpl.execute(WorkflowImpl.java:70)
	at org.jreleaser.maven.plugin.JReleaserPackageMojo.execute(JReleaserPackageMojo.java:78)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:370)
	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:351)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:215)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:171)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:163)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:294)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:960)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
Caused by: org.jreleaser.util.command.CommandException: Command execution error. exitValue = 125
	at org.jreleaser.packagers.AbstractPackagerProcessor.executeCommand(AbstractPackagerProcessor.java:238)
	... 39 more
```

**Comment by aalmiray** [comment]
FWIW the imageName is overwritten when the project is deemed as a snapshot

https://github.com/jreleaser/jreleaser/blob/d037c13d89c1dbda4e149783271e021a0590732d/core/jreleaser-model-impl/src/main/java/org/jreleaser/model/internal/validation/DockerPackagerValidator.java#L180-L186

**Comment by aalmiray** [comment]
Released in v1.3.0 -> https://github.com/jreleaser/jreleaser/releases/tag/v1.3.0
